code/modules/vehicles/multitile/multitile_verbs.dm

```diff
@@ -327,12 +327,87 @@
 	if(honk_sound)
 		playsound(loc, honk_sound, 75, TRUE, 15)	//heard within ~15 tiles
 
+//Finally, a broadcaster!
+/obj/vehicle/multitile/proc/use_broadcaster()
+	set name = "Use Broadcaster"
+	set desc = "Allows to send a loud message to the exterior via hull broadcaster."
+	set category = "Vehicle.General"
+
+	var/mob/living/user = usr
+	if(!istype(user))
+		return
+
+	var/obj/vehicle/multitile/V = user.interactee
+	if(!istype(V))
+		return
+
+	var/seat
+	for(var/vehicle_seat in V.seats)
+		if(V.seats[vehicle_seat] == user)
+			seat = vehicle_seat
+			break
+	if(!seat)
+		return
+
+	if(user.client)
+		if(user.client.prefs.muted & MUTE_IC)
+			to_chat(src, SPAN_DANGER("You cannot speak in IC (muted)."))
+			return
+	if(user.silent)
+		return
+
+	if(V.health < initial(V.health) * 0.5)
+		to_chat(user, SPAN_WARNING("\The [V]'s hull is too damaged to operate!"))
+
+	if(!V.broadcaster_available)
+		to_chat(user, SPAN_DANGER("Broadcaster was recently used. Wait a bit!"))
+		return
+
+	var/message = strip_html(input(user, "Enter a message to broadcast", "Vehicle broadcaster", null)  as text)
+	if(!message)
+		return
+
+	if(!V.broadcaster_available)
+		to_chat(user, SPAN_DANGER("Broadcaster was recently used. Wait a bit!"))
+		return
+
+	if(V.health < initial(V.health) * 0.5)
+		to_chat(user, SPAN_WARNING("\The [V]'s hull is too damaged to operate!"))
+
+	if(user.interactee != V)
+		return
+
+	//we simulate actually speaking into microphone, so those in interior can also hear it.
+	user.say(message)
+
+	//then we capitalize and proceed with bradcasting part
+	message = capitalize(message)
+
+	log_admin("[key_name(user)] used \a [V]'s broadcaster to say: >[message]<")
+
+	if(user.stat == CONSCIOUS)
+		var/list/mob/living/carbon/human/recipients = viewers(8, V) // slow but we need it
+		for(var/mob/living/carbon/human/O in recipients)
+			if(O.species && O.species.name == "Yautja") //NOPE
+				O.show_message("\The [V] broadcasts something, but you can't understand it.")
+				continue
+			O.show_message("\The <B>[V]</B> broadcasts, [FONT_SIZE_LARGE("\"[message]\"")]",2) // 2 stands for hearable message
+
+		for(seat in V.seats)
+			if(ishuman(V.seats[seat]))
+				recipients += V.seats[seat]
+
+		user.langchat_vehicle_broadcast(message, recipients, user.get_default_language(), V)
+
+		V.broadcaster_available = FALSE
+		addtimer(VARSET_CALLBACK(V, broadcaster_available, TRUE), 5 SECONDS)
```

Comment:
use COOLDOWN_DECLARE system here rather than callback timers for cooldowns

Reply:
Done.